pkg/sdk/resource_monitors.go

```diff
@@ -47,9 +173,18 @@ func (v *ResourceMonitor) ObjectType() ObjectType {
 
 // CreateResourceMonitorOptions contains options for creating a resource monitor.
 type CreateResourceMonitorOptions struct {
-	create          bool                    `ddl:"static" sql:"CREATE"`           //lint:ignore U1000 This is used in the ddl tag
+	create          bool                    `ddl:"static" sql:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace       *bool                   `ddl:"keyword" sql:"OR REPLACE"`
 	resourceMonitor bool                    `ddl:"static" sql:"RESOURCE MONITOR"` //lint:ignore U1000 This is used in the ddl tag
 	name            AccountObjectIdentifier `ddl:"identifier"`
+	With            *bool                   `ddl:"keyword" sql:"WITH"`
```

Comment:
With is a static value, i think. Its not a required or optional parameter, its just there.

Reply:
Well, if we create empty resource monitor, then the query won't compile(`create resource monitor 'foo' with` is not ok)

Reply:
Well, if we create empty resource monitor, then the query won't compile(`create resource monitor 'foo' with` is not ok)

Reply:
In that case would it make sense to have With be a separate type?

for example: 
```
With *ResourceMonitorWith `ddl:"keyword" sql:"WITH"`
```
and then later have the ResourceMonitorWith type declared which has all the parameters in it?

Reply:
ok